server/handler/export.go

```diff
@@ -0,0 +1,104 @@
+package handler
+
+import (
+	"database/sql"
+	"encoding/csv"
+	"net/http"
+	"strconv"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/src-d/gitbase-playground/server/serializer"
+	"github.com/src-d/gitbase-playground/server/service"
+)
+
+// Export returns a function that forwards an SQL query to gitbase and returns
+// the rows as CSV file
+func Export(db service.SQLDB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := func(w http.ResponseWriter, r *http.Request) error {
+			query := r.URL.Query().Get("query")
+			if query == "" {
+				return serializer.NewHTTPError(http.StatusBadRequest,
+					`Bad Request. Expected body: { "query": "SQL statement" }`)
+			}
+
+			rows, err := db.Query(query)
+			if err != nil {
+				return dbError(err)
+			}
+			defer rows.Close()
+
+			columnNames, columnTypes, err := columnsInfo(rows)
+			if err != nil {
+				return err
+			}
+
+			columnValsPtr := genericVals(columnTypes)
+
+			w.Header().Set("Content-Disposition", "attachment; filename=export.csv")
+			w.Header().Set("Content-Type", "text/csv")
+			csvWriter := csv.NewWriter(w)
+
+			csvWriter.Write(columnNames)
+
+			for rows.Next() {
+				if err := rows.Scan(columnValsPtr...); err != nil {
+					return err
+				}
+
+				record := make([]string, len(columnTypes))
+
+				for i, val := range columnValsPtr {
+					switch v := val.(type) {
+					case *sql.NullBool:
+						sqlVal, _ := val.(*sql.NullBool)
+						if sqlVal.Valid {
+							record[i] = strconv.FormatBool(sqlVal.Bool)
+						}
+					case *mysql.NullTime:
+						sqlVal, _ := val.(*mysql.NullTime)
+						if sqlVal.Valid {
+							b, err := sqlVal.Time.MarshalText()
+							if err != nil {
+								return err
+							}
+							record[i] = string(b)
+						}
+					case *sql.NullInt64:
+						sqlVal, _ := val.(*sql.NullInt64)
+						if sqlVal.Valid {
+							record[i] = strconv.FormatInt(sqlVal.Int64, 10)
```

Comment:
for example, why is this implementation different compared to [the other one in `handler.Query()`](https://github.com/src-d/gitbase-playground/blob/5967d42c9a4c206088d2be593ada537809a13dfc/server/handler/query.go#L111)?